live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/RepoEventFilterHandler.java

```diff
@@ -42,13 +50,65 @@
 @Slf4j
 public class RepoEventFilterHandler
 {
-
     private final List<RepoEventFilterApplier> repoEventFilterAppliers;
+    private final CamelEventMapper camelEventMapper;
+
+    /**
+     * Method handles event/node filtering. When node is denied by any of filters, a custom property "DENY_NODE: true" in Camel Exchange is created (used to filter out in main route). In case of update event: - when current version of the node is allowed but previous version was denied, then the type of the event is altered to 'Created'. - when current version of the node is denied but previous version was allowed, then the type of the event is altered to 'Deleted'.
+     *
+     * @param exchange
+     *            Camel Exchange Object
+     * @param filter
+     *            Filter configuration
+     */
+    public void handle(Exchange exchange, Filter filter)
+    {
+        final FilteringResults filteringResults = calculateFilteringResults(exchange, filter);
+        final boolean allowNode = filteringResults.allowed;
+        if (!allowNode)
+        {
+            exchange.setProperty(DENY_NODE, true);
+        }
+        filteringResults.newEventType.ifPresent(type -> exchange.getIn().setBody(camelEventMapper.alterRepoEvent(exchange, type)));
```

Comment:
WDYT about moving this step to `EventProcessor#proces()`?

Reply:
Actually, my previous PR has a similar part there.
We could possibly move all the filter related logic into that processor (as this single part would be quite hard to move - filtering results are `record`).
The question is whether we want to handle all of that in a single class (of course it would call/reuse classes with filtering logic)?
IMO, we should rather process in chunks, especially when we have `Camel` on board - but I generally don't think we are utilizing `Camel` well in the project and have spaghetti architecture because of that (another IMO - we may want to think about getting rid of it).
Anyway, I don't think it will be hard to move filtering logic to `EventProcessor` and reuse existing code if needed but for now I would leave that as is.

Reply:
I would leave it here, I was rather thinking about setting a property like in previous PR: `exchange.setProperty(UPDATED_EVENT_TYPE_PROP, NODE_CREATED.getType());` and do the altering in 'EventProcessor#proces()'